### DISCUSSION TITLE
Option to disable automatic search results in song select

### DISCUSSION THREAD

**Discussion by Flamiii** [ppy/osu#15277] [Ideas] [answered]
I have a ton of beatmaps loaded into the game (~23,000 in osu!lazer / ~83,000 in osu!stable) so this is probably why I'm experiencing so much lag in song select. Regardless, whenever I search for a beatmap in song select, my game will freeze for a few seconds and then it will display the results of my search. It's tolerable for me but it definitely can get kind of annoying, and I know many other people who not only have more beatmaps than me, but also have worse PCs.

I wouldn't really mind if it happened only once, but if I'm trying to search up a beatmap with a long name, it might end up lagging for several seconds multiple times before I even get the search results that I'm looking for and it can be kind of frustrating. I'd imagine that there's still a lot of room for optimization but I don't know when that optimization will be implemented.

https://user-images.githubusercontent.com/74088315/138611077-4ff2f324-d557-4311-bb75-89b1b3738cc0.mp4

In the meantime, a feature that causes the search results in song select to appear whenever I click the search icon or press the 'Enter' key instead of appearing automatically might help alleviate this issue, as I'd be able to type out the full name of a beatmap into the search box first and then press 'Enter' and only have my game freeze once as opposed to it lagging several times as I try to type in the name of the beatmap.

**Comment by bdach** [comment, marked as answer]
it is not the standard operating procedure here to implement workarounds for performance issues. this will be optimised properly.
